generator/src/main/java/com/scottlogic/deg/generator/generation/DecisionTreeDataGenerator.java

```diff
@@ -19,7 +19,7 @@
 
 public class DecisionTreeDataGenerator implements DataGenerator {
     private final DecisionTreeWalker treeWalker;
-    private final DataGeneratorMonitor monitor;
+    private final ReductiveDataGeneratorMonitor monitor;
```

Comment:
Can you leave this as the `DataGeneratorMonitor`, `startTimer` is on this interface.
Its not something that is specific to the reductive walker, all walker types can say when generation starts.